src/Raven.Server/Documents/Indexes/Workers/HandleSuggestions.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using Raven.Server.Config.Categories;
+using Raven.Server.Documents.Indexes.Persistence.Lucene;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Logging;
+
+namespace Raven.Server.Documents.Indexes.Workers
+{
+    public class HandleSuggestions : IIndexingWork
+    {
+        private readonly Logger _logger;
+
+        private readonly Index _index;
+        private HashSet<string> _suggestionsFields;
+        private readonly IndexingConfiguration _configuration;
+        private readonly DocumentsStorage _documentsStorage;
+        private readonly IndexStorage _indexStorage;
+
+        public HandleSuggestions(Index index, HashSet<string> suggestionsFields, DocumentsStorage documentsStorage, IndexStorage indexStorage, IndexingConfiguration configuration)
+        {
+            _index = index;
+            _configuration = configuration;
+            _suggestionsFields = suggestionsFields;
+            _documentsStorage = documentsStorage;
+            _indexStorage = indexStorage;
+            _logger = LoggingSource.Instance
+                .GetLogger<HandleReferences>(_indexStorage.DocumentDatabase.Name);
+        }
+
+        public string Name => "Suggestions";
+
+        public bool CanContinueBatch(DocumentsOperationContext documentsContext, TransactionOperationContext indexingContext, IndexingStatsScope stats, long currentEtag, long maxEtag)
```

Comment:
Probably, I dont know the code that much to being able to make that call at this point in time.